feat(trace-tree-node): Mitigating no instrumentation guards usage

Abdkhan14 opened the pull request:
- Currently we are checking to see if the node is a noInstrumentation Node or not, to determine the colour of the text inside the bar in the waterfall. This is very specific. 
- Added a `type="light" | "dark"` to the response of `node.makeBarColor` to be used in `virtualizedViewManager`. 

Only use of the this guard now is in the: 
- Test files to check for this node type specifically 
- and in `TraceTree.RemoveMissingInstrumentationNodes` method to filter out nodes from the tree when we toggle options

Commit:
feat(trace-tree-node): Mitigating no instrumentation guards usage

Commit:
feat(trace-tree-node): Mitigating no instrumentation guards usage

Commit:
feat(trace-tree-node): Fixing tests

Commit:
Merge branch 'abdk/trace-tree-node-attrs-getter' into abdk/trace-tree-node-missing-ins-guards

gggritso reviewed (commented):
🤔 is it possible to do add a getter specifically for the span text colour, rather than doing a conditional on the bar colour?

Commit:
feat(trace-tree-node): Addressing PR suggestions

Commit:
feat(trace-tree-node): Addressing PR suggestions

Commit:
feat(trace-tree-node): Addressing PR suggestions

gggritso reviewed (commented):
Makes sense, just one question!

Abdkhan14 committed:
feat(trace-tree-node): Mitigating SpanNode type guard usage (#104481)

- Added a `measurements` getter to the baseNode class and used it to
remove SpanNode guards
- Split the common traceSpanRow component that was being used by
multiple node classes to their own files

---------

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>
Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

Abdkhan14 commented on static/app/utils/analytics/tracingEventMap.tsx:
```diff
@@ -131,9 +131,6 @@ export type TracingEventParameters = {
     type: string;
-    next_op?: string;
-    parent_op?: string;
-    previous_op?: string;
```
@gggritso oops, just noticed this one, didn't think these attrs added any value ^
